### PR TITLE
The deterministic CFC property suite, and one derivation of the absence policy (CT-2178, phase 3)

### PR DESCRIPTION
**Pull request by bfollington**
Phase 3 of CT-2178. Turns the audit from a reader into an instrument: it produces evidence deliberately, in both directions, so the checks are exercised rather than merely present.

## The absence policy, collapsed to one derivation

`enforce-explicit` published `absenceBehavior: permissive-if-absent` while failing closed. The label was wrong, and it was wrong because the policy was encoded twice — the model-facing output path branched on the mode, and the capability snapshot described the mode separately.

Settled against the 239-run session corpus: **151 mediation-absence denials, zero permissive exposures, and 85 of the 191 runs that declared themselves permissive went on to deny.** The behavior conforms to AH-CFC-6; the descriptor did not.

`cfcAbsenceBehaviorForMode` is now the single derivation. Both enforcing modes map to `fail-closed-if-absent`, and three decision sites in `#modelFacingToolOutput` branch on a disposition derived from it. Behavior is unchanged, and `cfc-absence-policy.test.ts` pins it for every mode at all three sites.

## The property suite — `packages/cf-harness/test/cfc-properties/`

Each property runs a scripted adversarial episode against the real engine and prompt loop, writes artifacts, then runs the audit over its own artifacts and asserts the verdict. The checker is the assertion library.

| Property | What it establishes |
| --- | --- |
| P-deny-egress | The corpus's first label-driven release refusal — a `policyRefusal` naming `sink-ceiling`, with the offending atom and the input key that carried it, which AUD-16 counts |
| P-allow | The same flow without the labeled input releases; no check fails, and the exact set a one-run corpus cannot settle is asserted |
| P-influence | AUD-8 reaches `pass` — its first exercise against evidence rather than absence |
| P-prompt-authority | Five of the six `*_requires_direct_command` codes had appeared in no test |
| P-no-silent-fallback | An enforcing run whose mediation is unavailable stays enforcing |
| P-delegation | Which of the two readings of delegation we implement — see below |
| P-absence | Every mode's published behavior is the behavior it takes, anchored to AH-CFC-6 |

P-refuse-start, P-dial-order and P-posture-parity are established elsewhere and documented as such rather than copied — a second copy is a second encoding.

## What P-delegation answers — the evidence behind AH-CFC-12a

AH-CFC-12 binds **capabilities**; the CFC specification's §18.2.4.3 bounds **observation**. A live delegation, run against the real engine and prompt loop, says we implement both — but at different layers, and which layer does the work is the whole finding:

- **At the address layer we implement only the capability reading.** A token the delegation named is swapped for the **bare reference** in whatever the child dispatches. A child holding a shell reads through a bound handle exactly as the parent would. The handle is an indirection over an address, not a boundary around a value.
- **At the output layer we implement the observation reading.** Read from the child's own transcript, its unmediated read comes back as `cf-harness.observation-denied`, `reason: "not-observable"`, carrying an opaque handle. The child never saw the content.

**So §18.2.4.3 is satisfied, but not by the mechanism AH-CFC-12 describes.** What actually bounds observation is the mediation every tool output passes through, which AH-CFC-12 never mentions. An implementation that read our clause literally — bind the capabilities, hand over the handle, no output mediation — would satisfy AH-CFC-12 and leak.

That gap is now closed in the spec: **#6799 merged AH-CFC-12a**, which states the observation bound explicitly, so the clause no longer licenses that implementation. This property is the empirical evidence behind it, and `test/cfc-properties/delegation.test.ts` pins both layers so neither can quietly stop holding.

## Two findings, and independent corroboration of the diagnosis

Pointing the suite at fresh artifacts immediately failed AUD-3 and AUD-9. Both turned out to be the check overclaiming rather than the harness underdelivering — AUD-3 counted the `run-pattern-source` sidecar the harness writes for itself, and AUD-9 demanded an invocation context from a side effect that reaches the fabric rather than the substrate that mints one.

**A second thread reached the same conclusion on both findings independently, and landed it as #6789 (`871c7cc7ed`) while this branch was in review.** Two threads converging separately on "the check was overclaiming" for both findings is stronger corroboration than either alone, and worth recording before it reads as coincidence.

Its AUD-3 exemption is character-for-character this branch's under another name. Its AUD-9 is better: it derives which side effects could carry a context from the run's own recorded contexts, so there is no second encoding of which tools transport, and it unifies AUD-2 with AUD-9 so the two cannot describe the same effects differently. This branch dropped its own version of both in favour of it, and deleted the drift test that existed only to guard the duplication it had introduced.

## Two methodological rules, both written into the audit README

- **A consistency check cannot detect a consistent wrong answer.** Collapsing two encodings removes the disagreement a test was reading; after the absence collapse, reverting the label still passes the per-mode agreement test, and only the AH-CFC-6 anchor catches it. Every collapse must ship with an anchor outside the system.
- **Asserting the absence of failures asserts nothing on its own.** A test that filters a check's findings for failures and asserts the list is empty passes whether the check found nothing or never ran — the confusion `inconclusive` exists to prevent, reintroduced at the assertion layer. Five assertions in this branch had that shape; `checkThatRan` is the fix.

## CI

Nightly `.github/workflows/cfc-properties.yml`, not per-PR. Runs the suite into one artifact root, then the corpus audit over it — AUD-16's refusal count and AUD-18's uniformity are population questions a single run cannot answer.

`--fail-on warn` stays where it is. `--expected-failures` names every finding the suite is known to produce with the issue closing it; anything else fails the job, **and so does an entry that stopped occurring**, so a closed gap must take its entry with it and the list shrinking is the progress signal. The job is green today with four entries.

## Gates

`deno fmt --check`, `deno lint`, `deno task check`, `check-no-waitfor`, `check-docs`, `check-skill-facts`, `tasks/ci-workflow.test.ts`, and `deno test -A` in `packages/cf-harness` — all exit 0 under the pinned Deno 2.9.4, each read by exit code.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/cf-harness/audit/checks/structural.ts [hunk 1]
```diff
@@ -631,7 +669,8 @@ const decisionCoverage: AuditCheck = {
       for (const entry of run.toolOutputs.entries) {
         if (listed.has(entry.fileName)) continue;
         if (
-          isRecord(entry.value) && entry.value.type === SUBAGENT_RAW_RETURN_TYPE
+          isRecord(entry.value) && typeof entry.value.type === "string" &&
+          HARNESS_WRITTEN_ARTIFACT_TYPES.has(entry.value.type)
```
Comment (bfollington) [inline packages/cf-harness/audit/checks/structural.ts]: 🔴 **Major — the exemption trusts the unaccounted payload's discriminator.** AUD-3 skips any unlisted file whose parsed value merely says `type: "cf-harness.run-pattern-source"` (or the raw-return type), regardless of its filename or producer. I constructed an unlisted `stray-bash.json` with that discriminator and AUD-3 returned `pass`. The artifact being investigated cannot self-attest that it is harness bookkeeping. Please validate the producer-specific filename plus the full bookkeeping shape (and correlate its output id), or store bookkeeping outside `tool-outputs/`; add a negative test proving a stray tool output with a colliding `type` still fails.

File: packages/cf-harness/audit/checks/structural.ts [hunk 2]
```diff
@@ -1300,9 +1340,21 @@ const evidenceRetention: AuditCheck = {
       },
       {
         name: "a CFC invocation context",
-        held: contexts.length > 0 || effects.length === 0,
+        held: contexts.length > 0 || transported.length === 0,
```
Comment (bfollington) [inline packages/cf-harness/audit/checks/structural.ts]: 🔴 **Major — one context currently covers every transported effect.** This is existential over the whole run, not a join. I built a run with a covered `bash` activity and an uncovered `run_skill_script` activity; with the one bash context, AUD-9 returned `pass`. AUD-2 also cannot rescue this shape because it infers the transporting tool ids from the contexts that survived. Build expected `toolId:outputId` keys from every `sandboxTransportedSideEffects(run)` activity and require a matching context for each. The drift test establishes the current tool set, but not per-invocation retention.

File: packages/cf-harness/audit/expected-failures.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  threshold: FailOnThreshold,
+): ExpectedFailureReconciliation => {
+  const findings = results.filter((result) =>
+    verdictFailsThreshold(result.verdict, threshold)
+  );
+  const matched: CheckResult[] = [];
+  const unexpected: CheckResult[] = [];
+  const used = new Set<ExpectedFailure>();
+  for (const finding of findings) {
+    const entry = expected.find((candidate) => matches(candidate, finding));
```
Comment (bfollington) [inline packages/cf-harness/audit/expected-failures.ts]: 🔴 **Major — an entry can suppress findings outside the run shape it names.** `runShape` is prose only, and `.find()` reuses the same entry for unlimited findings. I reconciled two AUD-9 failures with the same message on different run ids against one entry naming only the first shape: both landed in `matched`, nothing was stale/unexpected, and `reconciliationFails()` returned false. `detail: ""` would be broader still. Make the run selector machine-readable and part of matching, validate a non-empty stable reason/detail, and enforce the intended cardinality (or declare an explicit count/category) so a new occurrence cannot borrow an old entry.

File: packages/cf-harness/test/cfc-properties/delegation.test.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+    it("refuses the unmediated observation the child made through it", async () => {
+      // And this is what keeps the unbounded address from being an unbounded
+      // observation: the child's read returns through the same mediation the
+      // parent's would, so under an enforcing profile it fails closed.
+      //
+      // An implementation that kept the capability binding and dropped this
+      // would satisfy AH-CFC-12 as written, and would leak.
+      const episode = await runDelegationEpisode({ reachForWithheld: true });
+
+      expect(episode.modelVisibleText).not.toContain("child-observed-secret");
```
Comment (bfollington) [inline packages/cf-harness/test/cfc-properties/delegation.test.ts:288]: 🔴 **Major — P-delegation still has negative-only assertions that can pass without the claimed event.** This test passes if the child result is silently dropped; `runDelegationEpisode()` already computes `childObservationDenied`, but no test asserts it. Likewise the return-boundary case at lines 273–276 scans the whole family transcript and loops over zero tokens successfully; it does not isolate or require a child-to-parent return. Assert `childObservationDenied === true`, then inspect the actual delegate result/parent-facing return and require the expected witness before asserting that no unauthorized token crossed.

File: packages/cf-harness/test/cfc-absence-policy.test.ts [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+        const observed = await runUnmediatedRead(mode);
+        expect(observed.denial).toEqual({
+          reason: "not-observable",
+          handleType: "cf-harness.opaque-handle",
+        });
+        expect(observed.disclosedContent).toBe(false);
+      }
+    });
+
+    it("answers a filesystem-status observation the same way in every mode", async () => {
```
Comment (bfollington) [inline packages/cf-harness/test/cfc-absence-policy.test.ts:399]: 🟡 This says it pins the two status-error call sites, but `runStatusObservation()` only invokes `read_file`. The shared disposition preserves `edit_file` today by direct comparison with the base code, but this test cannot catch the edit-specific branch/redaction drifting later. Add an `edit_file` status-error episode for all four modes (or parameterize the helper over both tools).

**Review by bfollington — COMMENTED** [review]
_cf-review via Codex GPT-5, on behalf of @bfollington._

Scope: PR 6791 at `6214a71` · deep-read of the four requested invariants and adjacent live docs · pinned Deno 2.9.4 focused suite: 8 tests / 58 steps passed. I also exercised three adversarial audit constructions; each reproduced the hole described inline.

Verified clean: the `unclearedObservationDisposition` mapping preserves the old `read_file` and `edit_file` status-error behavior for all four modes (`disabled` exposes, `observe` exposes + warns, both enforcing modes redact + deny). Deriving all three uncleared-observation sites from the absence dial is consistent with AH-CFC-6. `expected-failures.ts` also fails in the simple one-to-one uncovered and stale cases.

🔴 Blocking: the four correctness findings inline mean AUD-3, AUD-9, expected-failure reconciliation, and P-delegation can all report more assurance than the evidence supports.

🟡 Improvements: the new status test does not exercise `edit_file`, and the live audit documentation still describes the pre-fix AUD-3/AUD-9 results.

Verdict: blocking until the audit match/join boundaries and vacuous property assertions are tightened.

File: packages/cf-harness/README.md [hunk 6]
```diff
@@ -1939,6 +1939,93 @@ checking anyway, and the report says whose requirement they are.
...
+  `test/docker-runsc-sandbox.test.ts` covers both enforcing modes, each
+  transport missing on its own, and both present.
+- **P-dial-order** and **P-posture-parity** — `audit/test/seeded-violations.ts`
+  turns AUD-13 to `fail` and to `warn` on non-conforming matrix points, and
+  `audit/test/deployment.test.ts` covers AUD-18. These are the stronger form for
+  these two: a posture is a declaration, and a live episode can only produce
+  postures the harness is willing to emit, so it cannot reach a non-conforming
+  matrix point at all without the same seeded mutation.
+
+Two checks fail on every fresh run today, and P-allow asserts the exact set so
```
Comment (bfollington) [inline packages/cf-harness/README.md]: 🟡 This live documentation now describes the state before this PR's AUD-3/AUD-9 narrowing: P-allow currently asserts no failing check, and the expected-failures file no longer lists these two run-pattern findings. `ONBOARDING.md:575-590` and the PR description repeat the same obsolete `FAIL 2` / red-nightly claim. Please update all three so operators are not told to ignore findings that the current audit no longer emits.

File: packages/cf-harness/audit/expected-failures.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    "the issue each is tracked under. The job fails on any finding not listed",
+    "here, AND on any entry here that stopped occurring \u2014 a fixed gap must take",
+    "its entry with it, so the list cannot rot into a permanent excuse. The list",
+    "shrinking is the progress signal; it is never appended to without an issue.",
+    "",
+    "This is not a way to lower the bar. `--fail-on warn` stays where it is; an",
+    "entry names one check on one run shape, so a check that starts failing on a",
+    "shape nobody enumerated is still a red run."
+  ],
+  "expected": [
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/expected-failures.json:15]: Stale — this predates the rebase onto #6789. AUD-3 no longer fails on the source sidecar, so there is nothing to list; the nightly reconciles green today.

File: packages/cf-harness/audit/test/sandbox-transport-drift.test.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+import { describe, it } from "@std/testing/bdd";
+import { expect } from "@std/expect";
+import { fromFileUrl, join } from "@std/path";
+
+import { SANDBOX_TRANSPORTED_SIDE_EFFECT_TOOLS } from "../checks/structural.ts";
+
+const TOOLS_DIR = fromFileUrl(new URL("../../src/tools", import.meta.url));
+
+/** The `toolId` a tool module declares, where it declares one. */
+const declaredToolId = (source: string): string | undefined =>
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/test/sandbox-transport-drift.test.ts]: Stale — `sandbox-transport-drift.test.ts` was deleted with the duplication it guarded when this branch took #6789's AUD-9 in place of its own.

File: packages/cf-harness/README.md [hunk 9]
```diff
@@ -1939,6 +1939,93 @@ checking anyway, and the report says whose requirement they are.
...
+that neither can quietly become permanent:
+
+- **AUD-3** — `run_pattern` persists a `run-pattern-source` artifact into
+  `tool-outputs/` that the run report's `toolOutputs` never lists. The historic
+  corpus holds 534 `run_pattern` outputs and no source artifact, so the check
+  passed 234 runs that predate the artifact and fails the first run audited with
+  one.
+- **AUD-9** — an executed side effect wants a CFC invocation context, and
+  `run_pattern` reaches the fabric rather than the sandbox, so none is recorded
+  for it.
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/README.md]: Fixed — the paragraph was rewritten after the rebase and now says both checks were the check overclaiming and are fixed, and names what a permitted run genuinely cannot settle.

File: packages/cf-harness/audit/expected-failures.ts [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+  /** Findings an entry covers, which do not fail the run. */
+  matched: readonly CheckResult[];
+}
+
+const matches = (
+  entry: ExpectedFailure,
+  finding: CheckResult,
+): boolean =>
+  entry.checkId === finding.checkId && finding.message.includes(entry.detail);
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/expected-failures.ts]: Fixed — `assertUsableExpectedFailure` refuses an empty `detail` (and an empty `checkId`, `runs` or `issue`) at load, before any matching.

File: packages/cf-harness/test/cfc-properties/delegation.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/delegation.test.ts:288]: Fixed — `childObservationDenied` now reads the child's own `bash` message from its transcript, where the denial actually is, and the test asserts it.

File: packages/cf-harness/audit/checks/structural.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/checks/structural.ts]: Confirmed and filed as CT-2206 against merged code — `substrateReachingSideEffects` derives the transporting set from surviving contexts, so a tool that lost every context is classified host-side and never enters the join. Out of this PR's diff: `structural.ts` here is byte-identical to main.

[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/checks/structural.ts]: Same finding, filed as CT-2206 — the `falsifiedBy` wording goes with that fix. `structural.ts` is unchanged by this PR.

File: packages/cf-harness/README.md [hunk 11]
```diff
@@ -1939,6 +1939,93 @@ checking anyway, and the report says whose requirement they are.
...
+  turns AUD-13 to `fail` and to `warn` on non-conforming matrix points, and
+  `audit/test/deployment.test.ts` covers AUD-18. These are the stronger form for
+  these two: a posture is a declaration, and a live episode can only produce
+  postures the harness is willing to emit, so it cannot reach a non-conforming
+  matrix point at all without the same seeded mutation.
+
+Two checks fail on every fresh run today, and P-allow asserts the exact set so
+that neither can quietly become permanent:
+
+- **AUD-3** — `run_pattern` persists a `run-pattern-source` artifact into
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/README.md]: Out of scope — `run_pattern`'s piece-retention contract is not what this PR documents, and speculating it here would put a second description of that behaviour in a place nobody maintains.

File: .github/workflows/cfc-properties.yml [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        uses: ./.github/actions/deno-setup
+
+      - name: 🔍 Verify lock file & install dependencies
+        uses: ./.github/actions/deno-install
+
+      # One root for every property, so the corpus checks have a population.
+      - name: 🧪 Run the CFC property suite
+        working-directory: packages/cf-harness
+        env:
+          CF_HARNESS_PROPERTY_ARTIFACT_ROOT: ${{ runner.temp }}/cfc-properties
```
[the remark this answers is not in the document]
Comment (bfollington) [reply .github/workflows/cfc-properties.yml:56]: Accepted, not changed — Deno runs test files and steps sequentially without `--parallel`, and the episodes that share a run id write identical content, so last-write-wins costs repeated work rather than a wrong corpus. Worth revisiting if this suite ever runs parallel.

File: packages/cf-harness/test/cfc-properties/no-silent-fallback.test.ts [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+  /** The mode the run's own state records after the unmediated observation. */
+  recordedMode: string;
+  /** Whether the model-facing result was the typed denial. */
+  denied: boolean;
+  modelVisibleText: string;
+  reasonCodes: readonly string[];
+}
+
+const runUnmediatedEpisode = async (): Promise<FallbackEpisode> => {
+  const runId = "no-silent-fallback";
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/no-silent-fallback.test.ts:52]: Same as the workflow comment — sequential by default, and each of the four steps re-runs one deterministic episode. No interleaving is possible today.

File: packages/cf-harness/src/prompt-loop.ts [hunk 14]
```diff
@@ -1860,6 +1861,45 @@ const scrubBareFabricIdentifiersDeep = (value: unknown): unknown => {
...
+): UnclearedObservationDisposition => {
+  switch (cfcAbsenceBehaviorForMode(mode)) {
+    case "not-required":
+    case "permissive-if-absent":
+      return "expose";
+    case "observe-only":
+      return "expose-with-warning";
+    case "fail-closed-if-absent":
+      return "deny";
+  }
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/src/prompt-loop.ts:1966]: Valid and fixed — verified first that an unknown mode made the descriptor answer `undefined`, the disposition switch fall through, and the output be exposed. Both switches now default to the closed answer.

File: packages/cf-harness/audit/cli.ts [hunk 15]
```diff
@@ -339,6 +360,31 @@ export const runAuditCli = async (
...
+        }`,
+      );
+      return 2;
+    }
+    const reconciliation = reconcileExpectedFailures(
+      results,
+      file.expected ?? [],
+      options.failOn,
+    );
+    write(`\n${renderReconciliation(reconciliation)}`);
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/cli.ts]: Valid and fixed — `--json` now emits one document carrying `results` and `reconciliation`, and the human footer says the reconciliation decides the exit rather than promising a threshold exit it does not control.

File: packages/cf-harness/audit/cli.ts [hunk 16]
```diff
@@ -339,6 +360,31 @@ export const runAuditCli = async (
...
+    return reconciliationFails(reconciliation) ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/cli.ts]: Valid and fixed with the JSON one — the footer is now list-aware and no longer contradicts a green run that has findings in it.

File: packages/cf-harness/audit/cli.ts [hunk 17]
```diff
@@ -339,6 +360,31 @@ export const runAuditCli = async (
...
   );
+  if (options.expectedFailures !== undefined) {
+    // Held to a list rather than to a threshold. A finding no entry covers
+    // fails, and so does an entry that matched nothing — a closed gap takes
+    // its entry with it, which is what stops the list becoming an excuse.
+    let file: ExpectedFailuresFile;
+    try {
+      file = JSON.parse(
+        await Deno.readTextFile(options.expectedFailures),
+      ) as ExpectedFailuresFile;
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/cli.ts]: Valid and fixed — `readExpectedFailures` rejects a top level that is not an object and an `expected` that is not an array, so a malformed file exits 2 with a reason instead of reading as a strict empty list.

File: packages/cf-harness/audit/expected-failures.ts [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+ *
+ * Reads only the results it is given; it neither knows nor cares which paths
+ * produced them.
+ */
+export const reconcileExpectedFailures = (
+  results: readonly CheckResult[],
+  expected: readonly ExpectedFailure[],
+  threshold: FailOnThreshold,
+): ExpectedFailureReconciliation => {
+  const findings = results.filter((result) =>
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/expected-failures.ts]: Valid and fixed — an entry is now marked used whenever anything it names occurs at any verdict, and the threshold only selects which unmatched results are failures. A test covers the `--fail-on fail` case you describe.

File: packages/cf-harness/audit/cli.ts [hunk 19]
```diff
@@ -81,6 +94,7 @@ const USAGE = [
   "usage: cfc-audit <runDir | artifactRoot> [more paths...]",
   "                 [--json] [--fail-on fail|warn|inconclusive]",
   "                 [--corpus] [--expect-refusals]",
+  "                 [--expected-failures <path>]",
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/audit/cli.ts:100]: Valid and fixed — the module synopsis lists `--expected-failures`.

File: packages/cf-harness/test/cfc-properties/influence.test.ts [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+      const episode = await runInfluenceEpisode();
+
+      expect(episode.influencing).not.toContain("call-1:stderr");
+    });
+
+    it("leaves AUD-8 with no disagreement to report", async () => {
+      const episode = await runInfluenceEpisode();
+
+      const audit = await auditArtifacts(episode.runDir);
+      const disagreeing = checkThatRan(audit, "AUD-8").filter((finding) =>
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/influence.test.ts:129]: Valid and fixed — the test now requires AUD-8 to reach `pass` rather than merely not fail, so `inconclusive` cannot read as agreement.

File: packages/cf-harness/test/cfc-properties/influence.test.ts [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+  };
+};
+
+describe("cfc property: influence accumulation", () => {
+  describe("P-influence — a labeled observation the model read", () => {
+    it("accumulates as influence on the run's model context", async () => {
+      // AH-CFC-7, the positive direction.
+      const episode = await runInfluenceEpisode();
+
+      expect(episode.influencing).toContain("call-1:stdout");
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/influence.test.ts:96]: Valid and fixed — the positive property now asserts the observed line reached the model-facing transcript, so a regression dropping stdout while keeping the influence entry fails.

File: packages/cf-harness/test/cfc-properties/deny-egress.test.ts [hunk 22]
```diff
@@ -0,0 +1,221 @@
...
+ *
+ * `withSource` is the only difference between the two properties: it decides
+ * whether the labeled reference reaches the pattern's inputs.
+ */
+const runEpisode = async (
+  options: { withSource: boolean },
+): Promise<EpisodeOutcome> => {
+  const fabric = await createLabeledFabric("enforce-explicit");
+  const runId = `deny-egress-${options.withSource ? "refused" : "allowed"}`;
+  const artifactRoot = await propertyArtifactRoot(runId);
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/deny-egress.test.ts:67]: Valid and fixed — the artifact root is created before the fabric, so nothing that can throw sits between the fabric and the `try` that disposes it. Same in the delegation episode.

File: packages/cf-harness/test/cfc-properties/support/episode.ts [hunk 23]
```diff
@@ -0,0 +1,457 @@
...
+ *
+ * `cfcFlowLabels: "persist"` is what makes a label survive into the store and
+ * so what makes a release measurement have anything to measure; at the default
+ * rung the properties below would pass by finding nothing rather than by the
+ * gate admitting the flow.
+ */
+export const createLabeledFabric = async (
+  cfcEnforcementMode: CfcEnforcementMode = "enforce-explicit",
+): Promise<LabeledFabric> => {
+  const storage = StorageManager.emulate({ as: signer });
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/support/episode.ts:212]: Valid and fixed — teardown closes storage in a `finally`, so a runtime that throws on dispose no longer leaves it open.

File: packages/cf-harness/test/cfc-absence-policy.test.ts [hunk 24]
```diff
@@ -0,0 +1,396 @@
...
+  const result = await loop.runPrompt({
+    prompt: "Read a file.",
+    promptSlotBinding: directPromptSlotBinding,
+  });
+
+  const toolMessage = result.transcript.find((message) =>
+    message.role === "tool" && message.toolName === "read_file"
+  );
+  expect(toolMessage).toBeDefined();
+  const disclosedContent = toolMessage!.content.includes(SENSITIVE_PATH);
```
[the remark this answers is not in the document]
Comment (bfollington) [reply packages/cf-harness/test/cfc-absence-policy.test.ts:319]: Valid and fixed — the status path now asserts the positive direction too: `disabled` and `observe` must disclose the sensitive path, for both `read_file` and `edit_file`.

File: packages/cf-harness/audit/checks/structural.ts (hunk 1 above)
Comment (bfollington) [reply packages/cf-harness/audit/checks/structural.ts]: Confirmed by reading, and filed as CT-2206 rather than fixed here. Worth recording that this branch's own dropped version keyed on `entry.value.type` identically, so it is a finding against the shape both implementations reached rather than against #6789. `structural.ts` in this PR is byte-identical to main — `git diff origin/main` on it is empty.

File: packages/cf-harness/audit/checks/structural.ts (hunk 2 above)
Comment (bfollington) [reply packages/cf-harness/audit/checks/structural.ts]: Confirmed, and subtler than the line suggests, so recording the mechanism for CT-2206: the `toolId:outputId` join in `substrateEffectsMissingContext` is correct, but `substrateReachingSideEffects` derives the transporting set from `invocationContextsOf(run)`. A tool whose invocations all lost their context therefore never enters the set, so it is classified host-side and the join never sees it. Main's comment names the all-or-nothing limit; this per-tool case slips past it. Not fixed here — out of this PR's diff.

File: packages/cf-harness/audit/expected-failures.ts (hunk 3 above)
Comment (bfollington) [reply packages/cf-harness/audit/expected-failures.ts]: Fixed. Entries now carry `runs`, an anchored regular expression over the run id that is part of matching, so an entry cannot speak for a run it does not name; `assertUsableExpectedFailure` refuses an empty `checkId`, `runs`, `detail` or `issue` and an uncompilable `runs` at load. Your reproduction is now a test, alongside anchoring (`run-1` must not match `run-10`) and the legitimate alternation case. On cardinality I took the run selector rather than explicit counts: the suite's run count changes as properties are added, so a count would fail for the wrong reason. Say if you want counts as well.

File: packages/cf-harness/test/cfc-properties/delegation.test.ts (hunk 4 above)
Comment (bfollington) [reply packages/cf-harness/test/cfc-properties/delegation.test.ts:288]: Fixed, and you were right about more than the assertion. `childObservationDenied` was false — because it looked for the denial in the parent's `delegate_task` output, where it never appears. The denial is what the child was handed; the parent sees only the summary written afterwards. Read from the child's own transcript it is there: `cf-harness.observation-denied`, `not-observable`, with an opaque handle. So the property's claim held and its witness was in the wrong place. Both are now asserted, and the return-boundary case requires the return to be non-empty before scanning it.

File: packages/cf-harness/test/cfc-absence-policy.test.ts (hunk 5 above)
Comment (bfollington) [reply packages/cf-harness/test/cfc-absence-policy.test.ts:399]: Fixed — `runStatusObservation` is parameterized over `read_file` and `edit_file` and both run across all four modes. Finding the policy event by wording turned out to be its own trap: the two tools name different disclosures (`filesystem path/status` versus `content, digest, path, or status`), so the event is now matched by the tool that wrote it. The positive direction is asserted too, per cubic's P3 — `disabled` and `observe` must disclose the path, or a regression that redacted in every mode would pass.

File: packages/cf-harness/README.md (hunk 6 above)
Comment (bfollington) [reply packages/cf-harness/README.md]: Fixed. The paragraph now says both findings were the check overclaiming and both are fixed, and names what a permitted run genuinely cannot settle instead. The PR description is rewritten. `ONBOARDING.md:575-590` I have not touched — checking it now and will report; if it repeats the obsolete claim it needs the same correction.

Comment (bfollington) [reply packages/cf-harness/README.md]: Checked `ONBOARDING.md:576-590`: it is already current, not stale. #6789 updated it in the same change — it describes AUD-2 and AUD-9 both warning on a host-side root, calls that the documented reading rather than a defect, and states `FAIL 0 WARN 2` as the expected verdict line. That matches main's behaviour and this branch's. Nothing to correct there; the stale copies were the README paragraph and the PR description, both now fixed.
